src/NodeAnalyzer/ScopeAnalyzer.php

```diff
@@ -56,6 +56,13 @@ public function resolveScope(
 
         /** @var MutatingScope|null $parentScope */
         $parentScope = $parentNode->getAttribute(AttributeKey::SCOPE);
-        return $parentScope;
+        if ($parentScope instanceof MutatingScope) {
+            return $parentScope;
+        }
+
+        $scope = $this->scopeFactory->createFromFile($smartFileInfo);
```

Comment:
This would create a new scope disconnected from other scopes. That is wrong

The point of this is to keep working with previous scope in case of refreshed node.

Reply:
Ok 👍 I am closing it then

Reply:
Thanks :+1: 

I think we're crashing sometimes with missing scope, but we need to find out those cases and complete them in the start point. PHPStan skips some nodes to go with scope, and we need to find out those spaces.